File: leetcode/136-single-number/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int ans = 0;
        for (int x : nums) ans ^= x;
        return ans;
    }
};

int main() {
    Solution sol;
    vector<int> nums = {4, 1, 2, 1, 2};
    cout << sol.singleNumber(nums) << endl; // 4
    return 0;
}

File: leetcode/136-single-number/solution.py
from typing import List


class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        ans = 0
        for x in nums:
            ans ^= x
        return ans


if __name__ == "__main__":
    print(Solution().singleNumber([4, 1, 2, 1, 2]))  # 4